### PR TITLE
YLP-1114 Review routes and data structure for message-api

### PR DESCRIPTION
**Pull request by CedricPerotto**
New message-api service available as 1.0.0-rc1. See YourLoops updated config here:
https://github.com/mdblp/yourloops/pull/606

### CODE REVIEW

File: packages/yourloops/lib/data/api.ts [hunk 1]
```diff
@@ -319,18 +314,9 @@ export async function getMessageThread(session: Session, messageId: string): Pro
   });
 
   if (response.ok) {
-    const out = (await response.json()) as MessagesThread | undefined;
-    const messages: MessageNote[] = out?.messages ?? [];
-    if (!Array.isArray(messages)) {
-      log.error("Expected an array of messages", { messages });
-      Promise.reject(new Error("Invalid response"));
-    }
+    const messages = (await response.json()) as MessageNote[];
```
Comment (yvon-dblg) [inline packages/yourloops/lib/data/api.ts:317]: Can we trust this new service to return an array ?
We don't want to crash if it's not the case.

File: packages/yourloops/models/message.ts [hunk 2]
```diff
@@ -41,11 +41,4 @@ interface MessageNote {
...
 
-/**
- * An array of messages
- */
-interface MessagesThread {
-  messages?: MessageNote[];
-}
-
-export { MessageNote, MessagesThread };
+export { MessageNote };
```
Comment (yvon-dblg) [inline packages/yourloops/models/message.ts:44]: MessageNote structure do not seems to reflect well the go one:
See https://github.com/mdblp/message-in-a-bottle/blob/cbaad1b30094812cd47b7b840021ad8abccfdc87/schema/message.go#L6

```suggestion
export default MessageNote;
```

**Review by yvon-dblg — COMMENTED** [review]
Very quick & incomplete review
Just a first look

**Review by ydrisR — CHANGES_REQUESTED** [review]
OK but I agree with Yvon, it does not hurt to keep the array assertion.
And the changelog need to be updated with 2.0.6-rc0 so we can release it for system tests

File: packages/yourloops/lib/data/api.ts (hunk 1 above)
Comment (ydrisR) [reply packages/yourloops/lib/data/api.ts:317]: :+1:

Comment (CedricPerotto) [reply packages/yourloops/lib/data/api.ts]: Updated

**Review by ydrisR — APPROVED** [review]
LGTM

**Pull request merged**